framework/src/org/checkerframework/framework/type/QualifierHierarchy.java

```diff
@@ -564,15 +565,28 @@ public AnnotationMirror greatestLowerBound(
         }
     }
 
+    /**
+     *
+     * @deprecated use {@link #getCorrespondingAnnotation(Collection, AnnotationMirror)} instead
+     */
+    @Deprecated
     public AnnotationMirror findCorrespondingAnnotation(
             AnnotationMirror aliased, Collection<? extends AnnotationMirror> a) {
-        AnnotationMirror top = this.getTopAnnotation(aliased);
-        for (AnnotationMirror anno : a) {
-            if (this.isSubtype(anno, top)) {
-                return anno;
-            }
-        }
-        return null;
+        return getCorrespondingAnnotation(a, aliased);
+    }
+
+    /**
+     * Returns the annotation from the hierarchy that corresponds to {@code annotationMirror} from
+     * a set of annotations.
+     *
+     * @param annos The set of annotations.
+     * @param annotationMirror annotation that is in same hierarchy as the returned annotation
+     * @return AnnotationMirror in annos in the same hierarchy as annotationMirror
```

Comment:
Does it always return one of the elements of the set?  This `@return` clause suggests that.